### PR TITLE
fix(python, rust): `ORDER BY` on unselected columns

### PR DESCRIPTION
**Pull request by SeanTroyUWO**
Finally getting around to addressing an issue on the discord by a member named Iain. In most SQL implementations you can `ORDER BY` on columns which are not actively `SELECT`'ed. Finally created an implementation that satisfies all the requirements as I see them:
 - You can sort original columns which were not actively selected 
 - You can sort constructed or aliased columns
 - You can use expressions in the `ORDER BY`clause
 - No regressions with other functionality, like `GROUP BY`, `HAVING` or `DISTINCT`

I'm not sure how to link a Discord discussion to Github. If you scroll up in the SQL channel you should find us talking about this issue, and how it originally arose.  

I can elaborate on the mechanics behind this fix if needed. Maybe I should add some unit tests too.

### CODE REVIEW

**Comment by alexander-beedie** [comment]
>  Maybe I should add some unit tests too.

Yes please - a handful of other tests, in addition to the one you added, showing each of the new behaviours would be great (expressions in ORDER BY, etc) 👍

**Comment by alexander-beedie** [comment]
FYI: looking at the lint, I think you need to run `make requirements` in your local area to update ruff/etc 👍

**Comment by SeanTroyUWO** [comment]
> > Maybe I should add some unit tests too.
> 
> Yes please - a handful of other tests, in addition to the one you added, showing each of the new behaviours would be great (expressions in ORDER BY, etc) 👍

Added a bunch. Also had to make it more complicated because of the `EXCLUDE` clause which I didn't know existed until now. Not in postgres.

File: py-polars/polars/io/_utils.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def _is_glob_pattern(file: str) -> bool:
 
 def _is_local_file(file: str) -> bool:
     try:
-        next(glob.iglob(file, recursive=True))  # noqa: PTH207
```
Comment (SeanTroyUWO) [inline py-polars/polars/io/_utils.py:20]: Clippy? Idk how this got here.

**Comment by ritchie46** [comment]
Nice. Thanks a lot @SeanTroyUWO. Great addition.

**Pull request merged**